tests/robottelo/decorators/test_host.py

```diff
@@ -0,0 +1,166 @@
+        ]
+        self.assertEqual(u'RHEL7.2.1', host._cached_host_os_version())
+        self._command.assert_called_once_with('cat /etc/redhat-release')
+        self._command.return_value.stdout = [
+            u'Doesnt matter because because its cached'
+        ]
+        self.assertEqual(u'RHEL7.2.1', host._cached_host_os_version())
+        # if called more than once cache didn't worked
+        self._command.assert_called_once_with('cat /etc/redhat-release')
+
+    @mock.patch('robottelo.decorators.host.LOGGER')
+    def test_command_error(self, logger):
+        cmd = SSHCommandResult(
+            stdout=[],
+            stderr=u'bash: generate: command not found\n',
+            return_code=127, output_format=None
+        )
+        self._command.return_value = cmd
+
+        host._get_host_os_version()
+        self._command.assert_called_once_with('cat /etc/redhat-release')
+        logger.warning.assert_called_once_with(
+            u'Host version not available: %r' % cmd)
+
+    @mock.patch('robottelo.decorators.host.LOGGER')
+    def test_command_parsing_error(self, logger):
+        cmd = SSHCommandResult(
+            stdout=[u'Fedora release 23 (Twenty Three)'],
+            return_code=0
+        )
+        self._command.return_value = cmd
+
+        host._get_host_os_version()
+        self._command.assert_called_once_with('cat /etc/redhat-release')
+        logger.warning.assert_called_once_with(
+            u'Host version not available: %r' % cmd
+        )
+
+
+class SkipIfHostVersionIsUnavailableTestCase(TestCase):
+    """Tests for :func:`robottelo.decorators.host.skip_if_host_is` when host
+    version isn't available
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        down_versions = u'6 6.1 6.1.1 7 7.0'.split()
+        up_versions = u'7.1.1 7.2 7.2.1 8'.split()
+        cls._up_and_down_versions = tuple(
+            u'RHEL' + v for v in chain(down_versions, up_versions)
+        )
+        cls._cache_patcher = mock.patch(
+            'robottelo.decorators.host._cached_host_os_version'
+        )
+        cls._cached_function_mock = cls._cache_patcher.start()
+        cls._cached_function_mock.return_value = 'Not Available'
+        cls._settings_patcher = mock.patch(
+            'robottelo.decorators.host.settings'
+        )
+        cls._settings_patcher.start()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls._cache_patcher.stop()
+        cls._settings_patcher.stop()
+
+    def test_negative_skipping_with_single_version(self):
+        for single_version in self._up_and_down_versions:
+            @host.skip_if_os(single_version)
+            def dummy():
+                return True
+
+            self.assertTrue(dummy())
+
+    def test_negative_skipping_with_multiple_versions(self):
+        @host.skip_if_os(*self._up_and_down_versions)
+        def dummy():
+            return True
+
+        self.assertTrue(dummy())
+
```

Comment:
What about testing if this will actually skip if the version is on the skip_if_os arguments?